include/arch/arc/v2/irq.h

```diff
@@ -68,6 +69,79 @@ extern void z_irq_spurious(void *unused);
 })
 
 
+/**
+ * Configure a 'direct' static interrupt.
+ *
+ * When firq has no separate stack, it's not safe to call C ISR handlers.
+ * So for this case, the priority cannot be set to 0
+ *
+ * When firq has separate stack, the stack checking can raise stack check
+ * exception as sp is switched to firq's stack. So for this case, the priority
+ * cannot be set to 0
+ *
+ * Note that for the above cases, if application still wants to use firq by
+ * setting priority to 0. Application can call z_irq_priority_set again.
+ * Then it's left to application to handle the details of firq
+ *
+ * See include/irq.h for details.
+ * All arguments must be computable at build time.
+ */
+#define Z_ARCH_IRQ_DIRECT_CONNECT(irq_p, priority_p, isr_p, flags_p) \
+({ \
+	Z_ISR_DECLARE(irq_p, ISR_FLAG_DIRECT, isr_p, NULL); \
+	if (IS_ENABLED(CONFIG_ARC_FIRQ) && \
+	    !IS_ENABLED(CONFIG_ARC_FIRQ_STACK) && \
+	    priority_p == 0) { \
+		z_irq_priority_set(irq_p, priority_p + 1, flags_p); \
```

Comment:
ok , 1 is better

Reply:
Well and maybe in comments explain why we use 1.
Maybe something like that in comments:

There're 2 special cases when we don't want to use ARC FastIRQ (FIRQ) and so instead of requested priority 0 set it to the next - 1.
1. FIRQ doesn't have a separate stack (CONFIG_ARC_FIRQ_STACK = y), it's not safe to call C ISR handlers. *Even though it might be obvious to you today please explain why separate FIRQ stack is needed for C ISR handlers as reader might assume that normal IRQ stack could be used for FIRQs as well*

2. FIRQ has separate stack (CONFIG_ARC_FIRQ_STACK=y) but at the same time stack checking is enabled (CONFIG_ARC_STACK_CHECKING=y). Stack checking logic will raise stack check exception as SP is also banked and so it will be used from bank1.